controllers/progresses.js

```diff
@@ -107,8 +108,38 @@ const createProgress = async (req, res) => {
  */
 
 const getProgress = async (req, res) => {
+  const { dev, page = 0, size = 100, type, userId, taskId } = req.query;
   try {
     const data = await getProgressDocument(req.query);
+    if (dev) {
+      const paginatedProgressDocs = await getPaginatedProgressDocument(req.query);
```

Comment:
Are we not making these two calls for same request?

Reply:
The first call is used to figure out the total number of snapshots in the database so we can calculate the next and previous page URLs and show null when needed.